pkg/deploy/deploy.go

```diff
@@ -273,36 +265,39 @@ func deleteResource(ctx context.Context, cli client.Client, found *unstructured.
 	return nil
 }
 
-func createResource(ctx context.Context, cli client.Client, res *resource.Resource, owner metav1.Object) error {
+// CreateOrUpdateResource deals with both creationg and update of resources.
+// When resource not found, call createResource()
+// When resource found: (in order to call)
+// - skip update if it is OdhDashboardConfig CR
+// - call skipUpdateOnAllowlistedFields() to pass Allowlisted fields either not found annotation or is not "true"
+// - call performPatch() to reconcile.
+func CreateOrUpdateResource(ctx context.Context, cli client.Client, res *resource.Resource, found *unstructured.Unstructured, owner metav1.Object) error {
 	obj, err := conversion.ResourceToUnstructured(res)
 	if err != nil {
 		return err
 	}
-	if obj.GetKind() != "CustomResourceDefinition" && obj.GetKind() != "OdhDashboardConfig" {
-		if err := ctrl.SetControllerReference(owner, metav1.Object(obj), cli.Scheme()); err != nil {
-			return err
-		}
+
+	// Create case
+	if found == nil {
+		return createResource(ctx, cli, obj, owner)
 	}
-	return cli.Create(ctx, obj)
-}
 
-func updateResource(ctx context.Context, cli client.Client, res *resource.Resource, found *unstructured.Unstructured, owner metav1.Object) error {
-	// Skip ODHDashboardConfig Update
+	// Exception to skip ODHDashboardConfig CR reconcile
 	if found.GetKind() == "OdhDashboardConfig" {
 		return nil
 	}
-
-	// only reconcile whiltelistedFields if the existing resource has annoation set to "true"
-	// all other cases, whiltelistedfields will be skipped by ODH operator
-	if managed, exists := found.GetAnnotations()[annotations.ManagedByODHOperator]; !exists || managed != "true" {
+	// handl allowlisted fields with annoation
+	switch managed, exists := found.GetAnnotations()[annotations.ManagedByODHOperator]; {
```

Comment:
`found.GetAnnotations()` may return nil

Reply:
in this case( when no annotation set at all) , it falls into `case !exists` as _managed == "" and exists is false_,  right? 
which is the same: users can config these fields

Reply:
yeah sorry, I was thinking in could panic but in fact, you can dereference a map value even on nil ptr.
So wonder if you can just skip the check on `exists`, since in fact what matter is only `managed != "true"`

Reply:
But refactoring (changes without amending functionality) and changing the predicate are different things. Does it make sense to move the condition to own function?

Reply:
> but we cannot only check on managed != "true" case

but if the annotation does not exists, it will return an empty string which actually satisfies the  `managed != "true"`, isn't it ?

Reply:
i see what you mean now. yes, then it is either managed == true or  the rest (covers no annotation, this annotation not exist,  this annotation is set to non-true)